dashboard/src/pages/hardwareDetails/HardwareDetailsFilter.tsx

```diff
@@ -151,6 +161,17 @@ const HardwareDetailsFilter = ({
     [paramFilter],
   );
 
+  const handleSelectTree = useCallback(
+    (value: string) => {
+      const idx = Number(value.split('__')[1]);
+
+      if (treeIndexes.includes(idx))
+        setTreeIndexes(treeIndexes.filter(i => i !== idx));
+      else setTreeIndexes([...treeIndexes, idx]);
```

Comment:
Can you give it a try using react hook form here? I believe will save us from these types of logic, put a timebox and if it doesn't work you can get back to this